### PR TITLE
Add zone set to models level

### PR DESCRIPTION
**Pull request by ekoval**
Part of: https://github.com/yola/domainservice/issues/345

### CODE REVIEW

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: tests/models/test_zone.py [hunk 1]
```diff
@@ -73,3 +73,24 @@ def test_records_lists_records_by_name(self):
...
+
+
+class TestSetCNameZone(FakedServiceTestCase):
+    def setUp(self):
+        self.user = User.get(email='foo@example.net')
+        self.user._service._add_zone('example.net', 'example.net')
+        self.result = self.user.create_cname_zone(
+            'example.net', ['cname.example.net'], 'resolve-to.example.org')
+
+    def test_returns_zone_object(self):
```
Comment (RayeN) [inline tests/models/test_zone.py]: we don't return zone object anymore.

File: tests/models/test_zone.py [hunk 2]
```diff
@@ -73,3 +73,24 @@ def test_records_lists_records_by_name(self):
         self.assertIsInstance(self.zone.records, dict)
         self.assertIsInstance(self.zone.records['example.com'], list)
         self.assertIsInstance(self.zone.records['example.com'][0], Record)
+
+
+class TestSetCNameZone(FakedServiceTestCase):
+    def setUp(self):
+        self.user = User.get(email='foo@example.net')
+        self.user._service._add_zone('example.net', 'example.net')
```
Comment (RayeN) [inline tests/models/test_zone.py]: Why do we need this step?

File: tests/models/test_zone.py [hunk 3]
```diff
@@ -73,3 +73,24 @@ def test_records_lists_records_by_name(self):
         self.assertIsInstance(self.zone.records, dict)
         self.assertIsInstance(self.zone.records['example.com'], list)
         self.assertIsInstance(self.zone.records['example.com'][0], Record)
+
+
+class TestSetCNameZone(FakedServiceTestCase):
```
Comment (RayeN) [inline tests/models/test_zone.py]: I don't think this test is useful at all.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: tests/models/test_zone.py [hunk 4]
```diff
@@ -73,3 +73,24 @@ def test_records_lists_records_by_name(self):
...
+    def setUp(self):
+        self.user = User.get(email='foo@example.net')
+        #self.user._service._add_zone('example.org', 'example.org')
```
Comment (snitch) [inline tests/models/test_zone.py]: E265 block comment should start with '# '

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: pycloudflare/models.py [hunk 5]
```diff
@@ -72,6 +72,12 @@ def create_host_zone(self, name, jump_start=False):
 
         return zone
 
+    def create_cname_zone(self, zone_name, subdomains, resolve_to):
+        host_service = self.get_host_service()
+        result = host_service.zone_set(
+            zone_name, self.user_key, subdomains, resolve_to)
+        return result
```
Comment (ekoval) [inline pycloudflare/models.py]: We discussed this offline with Voffka, and decided to leave raw response here for now. The reason is that returned data is specific to CNAME zone, and we need a special model for this.

Comment (RayeN) [inline pycloudflare/models.py]: As an option - we can introduce lazy access to `Zone` attributes, and allow it to be initialized with either `zone_name` or optional data

Comment (stefanor) [inline pycloudflare/models.py]: OK, yes, it would need to be initialised differently, though.

**Pull request merged**